x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/ApiKeyBoolQueryBuilder.java

```diff
@@ -160,6 +165,31 @@ private static QueryBuilder doProcess(QueryBuilder qb, Consumer<String> fieldNam
                 newQuery.to(query.to()).includeUpper(query.includeUpper());
             }
             return newQuery.boost(query.boost());
+        } else if (qb instanceof final SimpleQueryStringBuilder simpleQueryStringBuilder) {
+            if (simpleQueryStringBuilder.fields().isEmpty()) {
+                simpleQueryStringBuilder.field("*");
+            }
+            // override lenient if querying all the fields, because, due to different field mappings,
+            // the query parsing will almost certainly fail otherwise
+            if (QueryParserHelper.hasAllFieldsWildcard(simpleQueryStringBuilder.fields().keySet())) {
+                simpleQueryStringBuilder.lenient(true);
+            }
+            Map<String, Float> requestedFields = new HashMap<>(simpleQueryStringBuilder.fields());
+            simpleQueryStringBuilder.fields().clear();
+            for (Map.Entry<String, Float> requestedFieldNameOrPattern : requestedFields.entrySet()) {
+                for (String translatedField : ApiKeyFieldNameTranslators.translatePattern(requestedFieldNameOrPattern.getKey())) {
+                    simpleQueryStringBuilder.fields()
+                        .compute(
+                            translatedField,
+                            (k, v) -> (v == null) ? requestedFieldNameOrPattern.getValue() : v * requestedFieldNameOrPattern.getValue()
+                        );
+                    fieldNameVisitor.accept(translatedField);
+                }
+            }
+            if (simpleQueryStringBuilder.fields().isEmpty()) {
+                simpleQueryStringBuilder.field("match_nothing_place_holder_field_name_wildcard*");
```

Comment:
No, it's not for test only.
I forgot to comment in the code...
If we let the `SimpleQueryStringBuilder#fields()` be empty here, it's eventually going to be interpreted as `*` https://github.com/elastic/elasticsearch/blob/293d0302fb5a32d2947e57cbfe54251c6f61a1a5/server/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java#L414 This means it's going to search on the other fields in the `.security` index, including the disallowed ones. This conspicuous pattern (a non-pattern should work too for this query type) is put in here in order to avoid this.

Reply:
Ah yes, because the default is `*`, that makes sense. Thanks for clarifying!

Reply:
Pushed [b73a13d](https://github.com/elastic/elasticsearch/pull/104132/commits/b73a13d660d858edbafb2d41c2450706d5fc14ca) to add a comment explaining it.

Reply:
Nifty! I'm wondering if we can be more explicit by returning a `MatchNoneQueryBuilder` here instead (it also avoids "pointless" match work on a field we know doesn't exist). 

(Note: I may be overlooking something here!)

Reply:
Good point! I'll try it, I think it should work! Thanks!

Reply:
Pushed [0a7b76e](https://github.com/elastic/elasticsearch/pull/104132/commits/0a7b76e8bbeb382e1dca0aee9c7ce9af3c197315) , I think it will pass the tests.